homeassistant/components/reolink/media_source.py

```diff
@@ -59,19 +59,31 @@ async def async_resolve_media(self, item: MediaSourceItem) -> PlayMedia:
         data: dict[str, ReolinkData] = self.hass.data[DOMAIN]
         host = data[config_entry_id].host
 
-        vod_type = VodRequestType.RTMP
-        if host.api.is_nvr:
-            vod_type = VodRequestType.FLV
+        def get_vod_type() -> VodRequestType:
+            if filename.endswith(".mp4"):
+                return VodRequestType.PLAYBACK
+            if host.api.is_nvr:
+                return VodRequestType.FLV
+            return VodRequestType.RTMP
+
+        vod_type = get_vod_type()
 
         mime_type, url = await host.api.get_vod_source(
             channel, filename, stream_res, vod_type
         )
         if _LOGGER.isEnabledFor(logging.DEBUG):
-            url_log = f"{url.split('&user=')[0]}&user=xxxxx&password=xxxxx"
+            url_log = url
+            if "&user=" in url_log:
+                url_log = f"{url_log.split('&user=')[0]}&user=xxxxx&password=xxxxx"
+            if "&token=" in url_log:
```

Comment:
```suggestion
            elif "&token=" in url_log:
```
Since the code does not look like it is intended to handle both `&user` and `&token` params in the string, then probably `elif` is appropriate here.